python/lsst/meas/extensions/piff/piffPsf.py

```diff
@@ -0,0 +1,99 @@
+# This file is part of meas_extensions_piff.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import pickle
+import numpy as np
+from lsst.afw.typehandling import StorableHelperFactory
+from lsst.meas.algorithms import ImagePsf
+from lsst.afw.image import Image
+from lsst.geom import Box2I, Point2I, Extent2I
+
+
+class PiffPsf(ImagePsf):
+    _factory = StorableHelperFactory(
+        "lsst.meas.extensions.piff.piffPsf",
+        "PiffPsf"
+    )
+
+    def __init__(self, width, height, piffResult):
+        assert width == height
+        ImagePsf.__init__(self)
+        self.width = width
+        self.height = height
+        self.dimensions = Extent2I(width, height)
+        self.piffResult = piffResult
+
+    # Storable overrides
+
+    def isPersistable(self):
+        return True
+
+    def _getPersistenceName(self):
+        return "PiffPsf"
+
+    def _getPythonModule(self):
+        return "lsst.meas.extensions.piff.piffPsf"
+
+    def _write(self):
+        return pickle.dumps((self.width, self.height, self.piffResult))
+
+    @staticmethod
+    def _read(pkl):
+        return PiffPsf(*pickle.loads(pkl))
+
+    # ImagePsf overrides
+
+    def __deepcopy__(self, meta=None):
+        return PiffPsf(self.dimensions.x, self.dimensions.y, self.PiffResult)
```

Comment:
That's an interesting point.  Stack `Psf` objects are supposed to be immutable, with a c++ `clone` method (which redirects to `__deepcopy__` in python) only available so that multiple caches can be used.  Of course you're right that if a user mutated `.piffResult` then the above wouldn't work for `__deepcopy__`.  We could make `._piffResult` "private", but I think even better would be to wrap `.piffResult` with a readonly property.

(Also, noticed that I mis-capitalized this!  So definitely an error on this line!)